File: app/src/main/java/com/syroniko/casseteapp/LogInSignUp/ResetPasswordActivity.java
package com.syroniko.casseteapp.LogInSignUp;

import android.content.res.ColorStateList;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.ViewCompat;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.syroniko.casseteapp.R;
import com.syroniko.casseteapp.firebase.Auth;
import com.syroniko.casseteapp.firebase.AuthCallback;

import org.jetbrains.annotations.Nullable;

public class ResetPasswordActivity extends AppCompatActivity {

    EditText resetEmailAddress;
    TextView sendResetTextView;
    String userEmail;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_password);

        resetEmailAddress = findViewById(R.id.reset_email_edittext);
        sendResetTextView = findViewById(R.id.send_reset_pass_textview);

        TextView tx = findViewById(R.id.recoveraccounString_resect_act);
        Typeface custom_font = Typeface.createFromAsset(getAssets(),  "fonts/montsextrathic.ttf");
        tx.setTypeface(custom_font);

        resetEmailAddress.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                ColorStateList colorStateList = ColorStateList.valueOf(Color.parseColor("#0fa59b"));
                ViewCompat.setBackgroundTintList(resetEmailAddress,colorStateList);
            }
        });

        sendResetTextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                userEmail = resetEmailAddress.getText().toString();
                if (userEmail.length() == 0){
                    Toast.makeText(ResetPasswordActivity.this, "Please fill your email address.", Toast.LENGTH_SHORT).show();
                    return;
                }

                Auth.INSTANCE.resetPassword(userEmail,
                        new AuthCallback() {
                            @Override
                            public void onSuccess(@Nullable String uid) {
                                Toast.makeText(ResetPasswordActivity.this, "Email sent. Check your inbox!", Toast.LENGTH_SHORT).show();
                            }

                            @Override
                            public void onFailure() {
                                Toast.makeText(ResetPasswordActivity.this, "A problem occurred while sending your email. Please try again.", Toast.LENGTH_SHORT).show();
                            }
                        });
            }
        });
    }
}

File: app/src/main/java/com/syroniko/casseteapp/LogInSignUp/ResetPasswordActivity.kt
package com.syroniko.casseteapp.LogInSignUp

import android.content.Context
import android.content.Intent
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import android.view.View.OnFocusChangeListener
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import com.syroniko.casseteapp.MainClasses.toast
import com.syroniko.casseteapp.R
import com.syroniko.casseteapp.firebase.Auth
import com.syroniko.casseteapp.firebase.AuthCallback
import kotlinx.android.synthetic.main.activity_reset_password.*

class ResetPasswordActivity : AppCompatActivity() {

    //TODO use hilt here with provider or it'll crash
    private lateinit var auth: Auth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reset_password)

        val tx = findViewById<TextView>(R.id.recoveraccounString_resect_act)
        val customFont = Typeface.createFromAsset(assets, "fonts/montsextrathic.ttf")
        tx.typeface = customFont

        resetEmailEditText.onFocusChangeListener = OnFocusChangeListener { _, _ ->
            val colorStateList = ColorStateList.valueOf(Color.parseColor("#0fa59b"))
            ViewCompat.setBackgroundTintList(resetEmailEditText, colorStateList)
        }

        sendResetPassTextView.setOnClickListener {
            val userEmail = resetEmailEditText.text.toString()

            if (userEmail.isEmpty()) {
                toast("Please fill your email address.")
                return@setOnClickListener
            }

            auth.resetPassword(userEmail,
                object : AuthCallback {
                    override fun onSuccess(uid: String?) {
                        toast("Email sent. Check your inbox!")
                    }

                    override fun onFailure() {
                        toast("A problem occurred while sending your email. Please try again.")
                    }
                }
            )
        }
    }

    companion object{
        fun startActivity(context: Context){
            val i = Intent(context, ResetPasswordActivity::class.java)
            context.startActivity(i)
        }
    }
}